lib/Optimizer/GraphOptimizer/ConstantFolding.cpp

```diff
@@ -113,6 +114,43 @@ void run(Backend &backend, CompiledFunction &compiledF,
   context.movePlaceholderBindings().release();
 }
 
+static bool isCanonical(const NodeValue &RN, Backend &backend, Node *clonedC,
+                        size_t idx) {
+  auto resultLayoutStr =
+      backend.getTensorLayoutRequirements().getNthResultLayoutRequirements(
+          clonedC, idx);
+  auto resultLayout = TensorLayoutDescription(resultLayoutStr);
+  auto &canInstance = CanonicalTensorLayout::getInstance();
+  auto default4DStr = canInstance.getDefaultNDLayout(4);
+  auto default4D = TensorLayoutDescription(default4DStr);
+  if (resultLayout.getDims().size() == 4 &&
+      !canInstance.isSatisfiedBy(RN.getType(), default4D, &resultLayout)) {
+    return false;
+  }
+  return true;
+}
+
+static void
+bailOnNonCanLayout(Function *constEvaluationF, Module &mod,
```

Comment:
bailOnNonCanLayout -> bailOnNonCanonicalLayout

And maybe move the comment before the function to explain what the function does.

Reply:
done - there's a large block of code inside that explains it, but I added a short description on top.